Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift

```diff
@@ -164,19 +185,47 @@ public class ChatChannelListController: DataController, DelegateCallable, DataSt
             log.error("Failed to perform fetch request with error: \(error). This is an internal error.")
         }
     }
+
+    private func channelBelongsToController(_ channel: ChatChannel, change: ListChange<ChatChannel>) -> Bool {
+        if let filter = filter {
+            return filter(channel)
+        }
+
+        // Given that at the moment some delegate methods are not yet removed, but some integrators are still using
+        // those, we need to keep using them for now.
+        // This block should be removed once `shouldAddNewChannelToList` and `shouldListUpdatedChannel` methods are
+        // fully removed.
+        let deprecatedFallback: () -> Bool? = { [unowned self] in
+            switch change {
+            case .insert:
+                return self.delegate?.controller(self, shouldAddNewChannelToList: channel)
+            case .update:
+                return self.delegate?.controller(self, shouldListUpdatedChannel: channel)
+            default:
+                return nil
+            }
+        }
```

Comment:
nit: no need to weakify, the lifetime of that closure won't exceed method's execution time
```suggestion
        let deprecatedFallback: () -> Bool? = {
            switch change {
            case .insert:
                return self.delegate?.controller(self, shouldAddNewChannelToList: channel)
            case .update:
                return self.delegate?.controller(self, shouldListUpdatedChannel: channel)
            default:
                return nil
            }
        }
```

Reply:
You're right, this is a leftover of something else I tried